thorlcr/thorutil/thbuf.cpp

```diff
@@ -1395,6 +1399,7 @@ class CSharedWriteAheadBase : public CSimpleInterface, implements ISharedSmartBu
         Owned<CRowSet> outputOwnedRows;
         CRowSet *rowSet;
         unsigned row, rowsInRowSet;
+        bool lastWasNull=false; // for sanity check only (there should never be two consequetive nulls)
```

Comment:
This should be set back to false in reset() - otherwise a splitter in a child query that has no records will probably trigger the failure.  Alternatively we could remove this code as discussed.